frontend/src/components/features/home/shared/generic-dropdown-menu.tsx

```diff
@@ -54,7 +54,16 @@ export function GenericDropdownMenu<T>({
 }: GenericDropdownMenuProps<T>) {
   const hasItems = filteredItems.length > 0;
   const showEmptyState = !hasItems && !stickyTopItem && !stickyFooterItem;
-
+  const getUniqueItemKey = (item: T): string => {
+    if (typeof item === "string") return item;
+    if (typeof item === "object" && item !== null && "id" in item) {
+      return String(item.id);
```

Comment:
Issue:
The check for "id" in item does not ensure that id is a valid primitive value suitable for use as a key.

Impact:
If id is undefined, null, or an object, the generated key may be invalid or misleading (e.g., "undefined" or "[object Object]").

Suggestion:
Explicitly validate that id is a string or number before using it as a key.